GooglePlayInstant/Editor/AndroidManifestUpdater.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Xml.Linq;
+using UnityEngine;
+#if UNITY_2018_1_OR_NEWER
+using UnityEditor;
+using UnityEditor.Android;
+#endif
+
+namespace GooglePlayInstant.Editor
+{
+    /// <summary>
+    /// This class has two modes of operation depending on the version of Unity.
+    ///
+    /// Pre-2018: Saves manifest changes to Assets/Plugins/Android/AndroidManifest.xml
+    ///
+    /// 2018.1+: Obtains the AndroidManifest.xml after it is fully merged but before the build occurs, and updates it
+    /// according to whether this is a Play Instant build.
+    /// </summary>
+    public class AndroidManifestUpdater
+#if UNITY_2018_1_OR_NEWER
```

Comment:
For my info, where is this macro defined?
Is this the best way to do things? Why not define an interface which has two implementations : < 2018 and 2018+
Then use macro at one place to decide which impl to use. 
Sprinkling macro everywhere is not maintainable IMO